src/graphics/mesh.rs

```diff
@@ -467,9 +471,7 @@ impl MeshBuilder {
         P: Into<mint::Point2<f32>> + Clone,
     {
         if points.len() < 3 {
-            return Err(GameError::LyonError(
-                "MeshBuilder::polygon() got a list of < 3 points".to_string(),
-            ));
+            panic!("MeshBuilder::polygon() got a list of < 3 points");
```

Comment:
I don't think panicking here is good, panicking within a library robs a user of the ability to manage it.

Reply:
checking the input length manually is trivial though. I do agree it might be better to keep as Err

Reply:
As nobbele said, this error is trivial and is well documented. I don't think an error is required here.

Reply:
I too, would like to keep the error for the reason @sokorototo cited.

Imagine some kind of particle system, where custom vertices are generated randomly and there's a small bug in the code that sometimes creates vertices with less than 3 points.

In the panic-case, if this bug isn't found early enough, then the whole app might panic and crash in production.

In the error-case, it would only crash if one was too lazy to handle the result properly and just unwrapped it. If one wasn't and handled the result as one would expect, then he'd probably do something like throwing away the error, only logging it perhaps, since the app can probably go on even if some particles somewhere are missing.

The panic may be well documented and somewhat trivial, but to me this is one of these situations that we luckily have `Result`s for in Rust, to force our users to at least stop for a moment and reflect on what is supposed to happen when this fails.